object-manipulation-3/main.js

```diff
@@ -0,0 +1,388 @@
+  {
+    rank: '5',
+    suit: 'clubs',
+    value: 5
+  },
+  {
+    rank: '6',
+    suit: 'clubs',
+    value: 6
+  },
+  {
+    rank: '7',
+    suit: 'clubs',
+    value: 7
+  },
+  {
+    rank: '8',
+    suit: 'clubs',
+    value: 8
+  },
+  {
+    rank: '9',
+    suit: 'clubs',
+    value: 9
+  },
+  {
+    rank: 'Jack',
+    suit: 'clubs',
+    value: 10
+  },
+  {
+    rank: 'Queen',
+    suit: 'clubs',
+    value: 10
+  },
+  {
+    rank: 'King',
+    suit: 'clubs',
+    value: 10
+  },
+  {
+    rank: 'Ace',
+    suit: 'diamonds',
+    value: 11
+  },
+  {
+    rank: '1',
+    suit: 'diamonds',
+    value: 1
+  },
+  {
+    rank: '2',
+    suit: 'diamonds',
+    value: 2
+  },
+  {
+    rank: '3',
+    suit: 'diamonds',
+    value: 3
+  },
+  {
+    rank: '4',
+    suit: 'diamonds',
+    value: 4
+  },
+  {
+    rank: '5',
+    suit: 'diamonds',
+    value: 5
+  },
+  {
+    rank: '6',
+    suit: 'diamonds',
+    value: 6
+  },
+  {
+    rank: '7',
+    suit: 'diamonds',
+    value: 7
+  },
+  {
+    rank: '8',
+    suit: 'diamonds',
+    value: 8
+  },
+  {
+    rank: '9',
+    suit: 'diamonds',
+    value: 9
+  },
+  {
+    rank: 'Jack',
+    suit: 'diamonds',
+    value: 10
+  },
+  {
+    rank: 'Queen',
+    suit: 'diamonds',
+    value: 10
+  },
+  {
+    rank: 'King',
+    suit: 'diamonds',
+    value: 10
+  },
+  {
+    rank: 'Ace',
+    suit: 'hearts',
+    value: 11
+  },
+  {
+    rank: '1',
+    suit: 'hearts',
+    value: 1
+  },
+  {
+    rank: '2',
+    suit: 'hearts',
+    value: 2
+  },
+  {
+    rank: '3',
+    suit: 'hearts',
+    value: 3
+  },
+  {
+    rank: '4',
+    suit: 'hearts',
+    value: 4
+  },
+  {
+    rank: '5',
+    suit: 'hearts',
+    value: 5
+  },
+  {
+    rank: '6',
+    suit: 'hearts',
+    value: 6
+  },
+  {
+    rank: '7',
+    suit: 'hearts',
+    value: 7
+  },
+  {
+    rank: '8',
+    suit: 'hearts',
+    value: 8
+  },
+  {
+    rank: '9',
+    suit: 'hearts',
+    value: 9
+  },
+  {
+    rank: 'Jack',
+    suit: 'hearts',
+    value: 10
+  },
+  {
+    rank: 'Queen',
+    suit: 'hearts',
+    value: 10
+  },
+  {
+    rank: 'King',
+    suit: 'hearts',
+    value: 10
+  },
+  {
+    rank: 'Ace',
+    suit: 'spades',
+    value: 11
+  },
+  {
+    rank: '1',
+    suit: 'spades',
+    value: 1
+  },
+  {
+    rank: '2',
+    suit: 'spades',
+    value: 2
+  },
+  {
+    rank: '3',
+    suit: 'spades',
+    value: 3
+  },
+  {
+    rank: '4',
+    suit: 'spades',
+    value: 4
+  },
+  {
+    rank: '5',
+    suit: 'spades',
+    value: 5
+  },
+  {
+    rank: '6',
+    suit: 'spades',
+    value: 6
+  },
+  {
+    rank: '7',
+    suit: 'spades',
+    value: 7
+  },
+  {
+    rank: '8',
+    suit: 'spades',
+    value: 8
+  },
+  {
+    rank: '9',
+    suit: 'spades',
+    value: 9
+  },
+  {
+    rank: 'Jack',
+    suit: 'spades',
+    value: 10
+  },
+  {
+    rank: 'Queen',
+    suit: 'spades',
+    value: 10
+  },
+  {
+    rank: 'King',
+    suit: 'spades',
+    value: 10
+  }
+];
```

Comment:
Instead of writing the entire card deck by hand, I would make an array of the possible cards and the possible suits, and use those arrays to generate the deck 🤓

Reply:
I added a createDeck function to create a deck from possible suits and ranks